frontend/components/MediaMap.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <div class="flex w-96 h-96 rounded-lg map border p-0 m-auto">
+    <div
+      id="map-div"
+      class="w-full h-full z-0 rounded-lg"
+      ref="map"
+      alt="Map displaying a pin at the location of this organisation or event."
+    ></div>
+
+    <p
+      class="text-2xl text-center z-10 m-auto align-middle p-5 text-light-cta-orange dark:text-dark-cta-orange"
+      :class="{
+        absolute: !errorOccurred,
+      }"
+      :key="rerenderKey"
+    >
+      {{ errorMessage }}
+      <br />
+      {{ sorryMessage }}
+    </p>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted } from "@vue/runtime-core";
+import { ref } from "vue";
+import "leaflet/dist/leaflet.css";
+import L from "leaflet";
+
+const props = defineProps<{
+  addresses: Array<string>;
+}>();
+const localePath = useLocalePath();
+
+const rerenderKey = ref(0);
+const map = ref();
+
+type Marker = {
+  address: string;
+  lat: number;
+  lon: number;
+};
+
+let errorOccurred: boolean = false;
+let errorMessage: string;
+let sorryMessage: string;
+
+function handleMapError(error: Error) {
+  console.error(error);
+  errorOccurred = true;
+
+  // TODO: More helpful and better looking error messages
+  errorMessage = "There was a problem with the map service.";
+  sorryMessage = "Sorry about that!";
+  rerenderKey.value += 1; // rerender the error div
+  map.value.style.opacity = 0;
+  map.value.style.position = "absolute";
+}
+
+function drawMap(avgLat: number, avgLon: number, markers: Array<Marker>) {
+  let mapOptions = {
+    center: [avgLat, avgLon],
+    zoom: 13,
+  };
+
+  let leafletMap: L.map = new L.map("map-div", mapOptions);
+
+  let layer = new L.TileLayer(
+    "http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+  );
+  leafletMap.addLayer(layer);
+  markers.map((marker: Marker) => {
+    let pin = L.marker([marker.lat, marker.lon], { title: marker.address }); // add location pin to map
+    pin.addTo(leafletMap);
+  });
+}
+
+/*
+    NOTE: Below is an exmaple of the code to use when the backend code 
```

Comment:
Thanks for this, @samcdonovan! Glad to have some reference for when we start making the shift here soon 😊